File: 0001-0500/0014/0014.longest-common-prefix.cpp
/*
 * @lc app=leetcode id=14 lang=cpp
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        int n = strs.size();
        if (n == 0)
        {
            return "";
        }

        string res = "";
        sort(begin(strs), end(strs));
        string a = strs[0];
        string b = strs[n - 1];

        for (int i = 0; i < a.size(); i++)
        {
            if (a[i] == b[i])
            {
                res += a[i];
            }
            else
            {
                break;
            }
        }

        return res;
    }
};
// @lc code=end

File: 0001-0500/0014/0014.longest-common-prefix.js
/*
 * @lc app=leetcode id=14 lang=javascript
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    if (strs === null || strs.length === 0) {
        return ""
    }
    for (let i = 0; i < strs[0].length; i++) {
        let c = strs[0].charAt(i)
        for (let j = 1; j < strs.length; j++) {
            if (i === strs[j].length || strs[j].charAt(i) !== c)
                return strs[0].substring(0, i)
        }
    }
    return strs[0];
};
// @lc code=end

